modules/jdbc/src/main/java/org/testcontainers/jdbc/ContainerDatabaseDriver.java

```diff
@@ -81,50 +70,32 @@ public synchronized Connection connect(String url, final Properties info) throws
             return null;
         }
 
+        ConnectionUrl connectionUrl = ConnectionUrl.newInstance(url);
+
         synchronized (jdbcUrlContainerCache) {
 
-            String queryString = "";
+            String queryString = connectionUrl.getQueryString().orElse("");
             /*
               If we already have a running container for this exact connection string, we want to connect
               to that rather than create a new container
              */
-            JdbcDatabaseContainer container = jdbcUrlContainerCache.get(url);
+            JdbcDatabaseContainer container = jdbcUrlContainerCache.get(connectionUrl.getUrl());
             if (container == null) {
-                /*
-                  Extract from the JDBC connection URL:
-                   * The database type (e.g. mysql, postgresql, ...)
-                   * The docker tag, if provided.
-                   * The URL query string, if provided
-                 */
-                Matcher urlMatcher = URL_MATCHING_PATTERN.matcher(url);
-                if (!urlMatcher.matches()) {
-                    throw new IllegalArgumentException("JDBC URL matches jdbc:tc: prefix but the database or tag name could not be identified");
-                }
-                String databaseType = urlMatcher.group(1);
-                String tag = urlMatcher.group(3);
-                if (tag == null) {
-                    tag = "latest";
-                }
-
-                queryString = urlMatcher.group(4);
-                if (queryString == null) {
-                    queryString = "";
-                }
 
-                Map<String, String> parameters = getContainerParameters(url);
+                Map<String, String> parameters = connectionUrl.getContainerParameters();
 
                 /*
                   Find a matching container type using ServiceLoader.
                  */
                 ServiceLoader<JdbcDatabaseContainerProvider> databaseContainers = ServiceLoader.load(JdbcDatabaseContainerProvider.class);
                 for (JdbcDatabaseContainerProvider candidateContainerType : databaseContainers) {
-                    if (candidateContainerType.supports(databaseType)) {
-                        container = candidateContainerType.newInstance(tag);
+                    if (candidateContainerType.supports(connectionUrl.getDatabaseType())) {
+                        container = candidateContainerType.newInstance(connectionUrl);
```

Comment:
is this mod to connectionUrl instead of 'tag' intentional?

Reply:
Yes, it is intentional and part of the implementation. Please see PR Description for details.